### PR TITLE
CCD-2779 - Update validators to allow InjectedData in CaseEvent tab

### PR DESCRIPTION
**Pull request by sullivangareth**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CCD-2779

### Change description ###

Update validators to allow InjectedData in EventEnablingCondition column of CaseEvent tab

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -267,7 +267,7 @@ subprojects { subproject ->
         testCompile group: 'org.powermock', name: 'powermock-api-mockito2', version: powermockVersion
         testCompile group: 'org.powermock', name: 'powermock-module-junit4', version: powermockVersion
 
-        testCompile group: 'com.github.hmcts', name: 'ccd-test-definitions', version: '7.16.1'
+        testCompile group: 'com.github.hmcts', name: 'ccd-test-definitions', version: '7.18.0_CCD-2779'
```
Comment (sullivangareth) [inline build.gradle:270]: Currently pointing to pre-release, will need to be bumped to 7.18.0 once ccd-test-definitions PR is merged to `master`

Comment (sullivangareth) [reply build.gradle:270]: https://github.com/hmcts/ccd-test-definitions/pull/46

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/event/AbstractShowConditionValidator.java [hunk 2]
```diff
@@ -49,7 +63,8 @@ protected void validateShowConditionFields(EventEntity eventEntity,
                 if (!forShowConditionFieldExistsAtLeastOneCaseFieldEntity(
                     showConditionField,
                     eventEntity.getCaseType().getCaseFields())
-                    && !MetadataField.isMetadataField(showConditionField)) {
+                    && !(MetadataField.isMetadataField(showConditionField)
+                            || isInjectedField(showConditionField, isSupportingInjectedData))) {
```
Comment (danlysiak) [inline domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/event/AbstractShowConditionValidator.java:67]:
```suggestion
                    && !MetadataField.isMetadataField(showConditionField) 
                    && !isInjectedField(showConditionField, isSupportingInjectedData)) {
```
`!(A || B)` = `!A && !B`, I think the latter is a bit more readable!

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/event/AbstractShowConditionValidator.java [hunk 3]
```diff
@@ -62,6 +77,14 @@ protected void validateShowConditionFields(EventEntity eventEntity,
         }
     }
 
+    private boolean isInjectedField(String showConditionField, boolean isSupportingInjectedData) {
+        if (isSupportingInjectedData) {
+            return InjectedField.isInjectedField(showConditionField);
+        }
+
+        return false;
```
Comment (danlysiak) [inline domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/event/AbstractShowConditionValidator.java:85]:
```suggestion
        return isSupportingInjectedData && InjectedField.isInjectedField(showConditionField);
```

**Review by danlysiak — APPROVED** [review]
Approved with couple of optional comments

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/event/AbstractShowConditionValidator.java (hunk 2 above)
Comment (sullivangareth) [reply domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/event/AbstractShowConditionValidator.java:67]: Sure - makes sense - I had copied the approach from elsewhere

Will fix

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/event/AbstractShowConditionValidator.java (hunk 3 above)
Comment (sullivangareth) [reply domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/event/AbstractShowConditionValidator.java]: Fixed in https://github.com/hmcts/ccd-definition-store-api/pull/1203/commits/1a9c18091b83dd43eaf785f1bb6543fe7b24d22c

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/event/AbstractShowConditionValidator.java (hunk 2 above)
Comment (sullivangareth) [reply domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/event/AbstractShowConditionValidator.java]: Fixed in https://github.com/hmcts/ccd-definition-store-api/pull/1203/commits/1a9c18091b83dd43eaf785f1bb6543fe7b24d22c

**Review by CameronGuthrie1 — APPROVED** [review]

**Pull request merged**